classes/sota_freedom-u540.bbclass

```diff
@@ -0,0 +1,18 @@
+# Support fitImage by default
+KERNEL_CLASSES_sota = " kernel-fitimage "
+KERNEL_IMAGETYPE_sota = "fitImage"
+KERNEL_IMAGETYPES_remove_sota = "uImage"
+OSTREE_KERNEL = "${KERNEL_IMAGETYPE}-${INITRAMFS_IMAGE}-${MACHINE}-${MACHINE}"
+
+OSTREE_BOOTLOADER ?= "u-boot"
+INITRAMFS_FSTYPES = "cpio.gz"
+PREFERRED_PROVIDER_virtual/bootloader_sota ?= "u-boot"
+
+IMAGE_BOOT_FILES_sota ?= "fw_payload.bin boot.scr uEnv.txt"
+KERNEL_DEVICETREE_sota ?= "sifive/${RISCV_SBI_FDT}"
```

Comment:
Can we add `IMAGE_FSTYPES_remove = "wic.gz"` in here so we don't need it in the meta-updater-riscv layer?

Reply:
Argh, that doesn't work if you are using a fitImage

Reply:
So, there is actually no need to remove wic.gz, and I tested with it as well (flashed the wic.gz image into my sdcard). Which error are you getting with it?

Reply:
I see this error:

> ERROR: Task do_image_wic in /scratch/alistair/yocto/oe-master/meta/recipes-core/images/core-image-minimal.bb depends upon non-existent task IMAGE_CMD_ota-ext4 in /scratch/alistair/yocto/oe-master/meta/recipes-core/images/core-image-minimal.bb
> ERROR: Command execution failed: 1

I suspect you don't see if as you have the meta-updater-riscv layer which contains this line:
`IMAGE_FSTYPES_remove = "wic wic.gz wic.bmap ext4 ext4.gz"`
in
`recipes-core/images/initramfs-ostree-image.bbappend`.

I am currently not using they layer and was hoping to contain everything in the meta-riscv and meta-updater layer.

Reply:
Right, that is another pending fix that I will propose to meta-updater directly, but the issue is that the initramfs recipe is being contaminated by IMAGE_FSTYPES (which is also populated by the board config).

Reply:
Yep, I think your bbappend can just be put in the main recipe. That should fix the issue.

Reply:
There is no need for initramfs wic/ext4 images, which is why the error.